src/components/button/button.tsx

```diff
@@ -13,25 +15,41 @@ export type ButtonProps = {
   handleClick?: () => void;
   danger?: boolean;
   size?: ButtonSizes;
+  icon?: iconType;
+  iconOnRight?: boolean;
 } & StitchesButtonProps;
 
 const IonButton = ({
   type = 'primary',
   disabled = false,
   danger = false,
   size = 'md',
+  icon,
+  iconOnRight,
   label,
   handleClick,
 }: ButtonProps) => {
+  const iconSize = {
+    sm: 16,
+    md: 20,
+    lg: 24,
+    xl: 28,
+  };
+
   return (
     <ButtonStyles
       type={type}
       onClick={handleClick}
       disabled={disabled}
       danger={danger}
       size={size}
+      withIcon={icon && true}
+      iconOnRight={iconOnRight}
     >
-      {label}
+      <div>
+        {icon && <IonIcon type={icon} size={iconSize[`${size}`]} />}
```

Comment:
```suggestion
        {icon && <IonIcon type={icon} size={iconSize[`${size}`]} />}
```

https://itnext.io/7-bad-code-smells-in-typescript-e1cb397723c6

check topic 4.

Reply:
https://pt-br.reactjs.org/docs/conditional-rendering.html#inline-if-with-logical--operator